Src/BootCamp.Chapter/LinqMethodDemo.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    public static class LinqMethodDemo
+    {
+        internal static IEnumerable<T> Demo<T>(this IEnumerable<T> list, Predicate<T> predicate)
+        {
+            list = Filter(list, predicate);
+            list = ReduceToHalfLength(list);
+
+            return list; 
+        }
+
+        private static List<T> ReduceToHalfLength<T>(IEnumerable<T> list)
+        {
+            var indexOfHalfList = Math.Floor((decimal) list.Count() / 2);
+            var convertedToList = list.ToList();
+            var halfList = new List<T>(); 
+            // I start the loop one after the half of the list so I can remove all the items needed 
+            for (int i = 0 ; i < indexOfHalfList; i++)
+            {
+                halfList.Add(convertedToList[i]); 
+            }
+
+            return halfList; 
+        }
+
+        private static IEnumerable<T> Filter<T>(IEnumerable<T> list, Predicate<T> predicate)
```

Comment:
This method is not needed. It's done for you in LINQ. You can simply:
``list.Where(predicate)``. So remove this method and use the suggested LINQ instead.

Reply:
oke
then I have to look how to solve this error message : 
```
Argument 2: cannot convert from 'System.Predicate<T>' to 'System.Func<T, bool>'	
```

Reply:
Then that is your solution, the second one. Though I find it strange thst it is not a predicate.

Reply:
me too. but on the first one I see a error message.